lex/gen.go

```diff
@@ -1194,10 +1194,12 @@ func (ts *TypeSchema) writeTypeDefinition(name string, w io.Writer) error {
 	case "boolean":
 		pf("type %s bool\n", name)
 	case "object":
-		if len(ts.Properties) == 0 {
-			pf("type %s interface{}\n", name)
-			return nil
-		}
+		/*
+			if len(ts.Properties) == 0 {
+				pf("type %s interface{}\n", name)
+				return nil
+			}
+		*/
```

Comment:
was this intentional? if we'll come back to this later, a comment describing why it is commented out for now

Reply:
yeah, this was actually not used for a long while, and now that it is again, we actually want empty structs defined (the threadgate rules are all empty structs with a type field)